cpp/src/io/parquet/page_data.cu

```diff
@@ -1908,8 +1908,10 @@ void __host__ DecodePageData(hostdevice_vector<PageInfo>& pages,
   dim3 dim_block(block_size, 1);
   dim3 dim_grid(pages.size(), 1);  // 1 threadblock per page
 
-  gpuDecodePageData<<<dim_grid, dim_block, 0, stream.value()>>>(
-    pages.device_ptr(), chunks, min_row, num_rows);
+  if (pages.size() > 0) {
```

Comment:
```suggestion
  if (pages.size() > 0) {
    dim3 dim_block(block_size, 1);
    dim3 dim_grid(pages.size(), 1);  // 1 threadblock per page
```
...and also remove the code in lines above.

Reply:
Or another way:
```
if (pages.size() == 0) { return; }

dim3 dim_block(block_size, 1);
dim3 dim_grid(pages.size(), 1);  // 1 threadblock per page

gpuDecodePageData<<<dim_grid, dim_block, 0, stream.value()>>>(
      pages.device_ptr(), chunks, min_row, num_rows);
```

Reply:
@ttnghia can we return from `decode_page_data` when `pages.empty()`? Do we really need to to through whole (well, most of) `decode_page_data` when there's no data?

Reply:
Maybe this fix can get added to https://github.com/rapidsai/cudf/pull/11867 and I'll just close this PR ?

Reply:
Yeah then I'll add this into #11867. Thanks.

Reply:
Please see https://github.com/rapidsai/cudf/pull/11867/commits/36c7ec256743e8aa5e58eb62caf529d065d32e48. I've added assertion checks instead of silently exiting the function.

Reply:
works for me